From e6e0fb82261340f9181d1b2b11a9706a16c5dbc7 Mon Sep 17 00:00:00 2001
From: Zoltan Varga <vargaz@gmail.com>
Date: Fri, 4 Aug 2017 11:39:28 -0400
Subject: [PATCH] [btls] Disable BUILD_SHARED_LIBS while building boringssl, so
 its symbols are not exported from the btls shared lib. (#5307)

---
 mono/btls/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mono/btls/CMakeLists.txt b/mono/btls/CMakeLists.txt
index 9f2365d3e50..d33038d75cf 100644
--- a/mono/btls/CMakeLists.txt
+++ b/mono/btls/CMakeLists.txt
@@ -31,7 +31,10 @@ set (CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${BTLS_CFLAGS}")
 set (CMAKE_MACOSX_RPATH 1)
 set (MONO_BTLS 1)
 
+set(BUILD_SHARED_LIBS_SAVED "${BUILD_SHARED_LIBS}")
+set(BUILD_SHARED_LIBS OFF)
 add_subdirectory (${BTLS_ROOT} boringssl)
+set(BUILD_SHARED_LIBS "${BUILD_SHARED_LIBS_SAVED}")
 
 include_directories (
 	${SRC_DIR}
-- 
2.11.0

